src/adu_workflow/src/agent_workflow.c

```diff
@@ -1208,6 +1209,8 @@ static void CallDownloadHandlerOnUpdateWorkflowCompleted(const ADUC_WorkflowHand
                 workflow_add_erc(workflowHandle, result.ExtendedResultCode);
             }
         }
+
+        ADUC_FileEntity_Uninit(&fileEntity);
```

Comment:
Line 1198 is already uninitializing the fileEntity.

Reply:
fixed